lib/runtime/juttle-utils.js

```diff
@@ -73,6 +75,61 @@ function filterPoints(points, from, to) {
     });
 }
 
+
+function _setTimeField(pt, timeField) {
+    if (timeField) {
+        if (!pt[timeField]) {
+            this.trigger('warning', new errors.runtimeError('RT-POINT-MISSING-TIME', {field: timeField}));
+        } else if (timeField !== 'time') {
+            pt.time = pt[timeField];
+            delete pt[timeField];
+        }
+    }
+}
+
+function _ensureJuttleMoment(pt, epsilon) {
+    if (pt.time !== undefined && !(pt.time instanceof JuttleMoment)) {
+        var date = makeDate(pt.time);
+        if (date.getTime() === date.getTime()) { // speedy check for invalid date, to make sure its getTime() isn't NaN
+            pt.time = new JuttleMoment({rawDate: date});
+        } else {
+            try {
+                pt.time = new JuttleMoment(pt.time);
+            } catch(e) {
+                if (this.trigger) {
+                    this.trigger('warning', new errors.runtimeError('RT-INVALID-TIME-ERROR', {time: pt.time}));
+                }
+                return null;
+            }
+        }
+
+        if (epsilon) {
+            pt.time.epsilon = epsilon;
+        }
+    }
+    return pt;
+}
+
+// Assigns the time from timeField to 'time' and attempts to convert it
+// into JuttleMoment
+//
+// If not specified, timeField defaults to 'time'. If specified and point
+// is missing that field, emits a warning using the supplied context.
+function toNativeTime(points, options) {
```

Comment:
I feel like explicitly passing in the context is clearer than this approach.

I also think it makes total sense to add `AdapterRead.toNativeTime(points)` that just calls `toNativeTime(points, this.options, this)` so that in most of the cases we don't need to worry about it either way.